sqlx-core/src/net/tls/tls_native_tls.rs

```diff
@@ -61,7 +62,9 @@ pub async fn handshake<S: Socket>(
         builder.identity(identity);
     }
 
-    let connector = builder.build().map_err(Error::tls)?;
+    let connector = rt::spawn_blocking(move || builder.build())
```

Comment:
```suggestion
    // The openssl TlsConnector synchronously loads certificates from files. 
    // Loading these files can block for tens of milliseconds.
    let connector = rt::spawn_blocking(move || builder.build())
```

Just copied your description as a comment so we remember why this exists.